rsocket-core/src/main/java/io/rsocket/plugins/PluginRegistry.java

```diff
@@ -0,0 +1,57 @@
+package io.rsocket.plugins;
+
+import io.rsocket.DuplexConnection;
+import io.rsocket.RSocket;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PluginRegistry {
+  private List<DuplexConnectionInterceptor> connections = new ArrayList<>();
+  private List<RSocketInterceptor> clients = new ArrayList<>();
+  private List<RSocketInterceptor> servers = new ArrayList<>();
+
+  public PluginRegistry() {}
+
+  public PluginRegistry(PluginRegistry defaults) {
+    this.connections.addAll(defaults.connections);
+    this.clients.addAll(defaults.clients);
+    this.servers.addAll(defaults.servers);
+  }
+
+  public void addConnectionPlugin(DuplexConnectionInterceptor interceptor) {
+    connections.add(interceptor);
```

Comment:
It doesn't look like differentiating between different server/clients but applies the same interceptor list to all server and clients..